lib/domevent.ts

```diff
@@ -126,27 +126,57 @@ export function onMatch(selector: string, eventType: string, callback: EventCB,
   return (elem) => { new DomEventMatchListener(elem, eventType, callback, useCapture, selector); };
 }
 
+export type KeyEventType = 'keypress' | 'keyup' | 'keydown';
+
+export interface IKeyHandlers {
+  [key: string]: (this: void, ev: KeyboardEvent, elem: Element) => void;
+}
+
 /**
- * Listen to key presses, with specified per-key callbacks. The `onKeyPress()` variant takes no
- * `elem` argument, and may be used as an argument to dom().
- *
+ * Listen to key events (typically 'keydown' or 'keypress'), with specified per-key callbacks.
  * Key names are listed at https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values
  *
+ * Methods onKeyPress() and onKeyDown() are intended to be used as arguments to dom().
+ *
+ * By default, handled events are stopped from bubbling with stopPropagation() and
+ * preventDefault(). If, however, you register a key with a "$" suffix (i.e. "Enter$" instead of
```

Comment:
Is this convention mimicking another library or is it a grainjs special?

Reply:
No, I came up with it. It seems to me that normally if you handle a key event, you don't want it to bubble, but sometimes you need to opt out. This convention makes it easy -- I picked the "$" suffix because it doesn't even need to be quoted. Admittedly arcane, but I all alternatives I could think of were either significantly more verbose or equally arcane, so I settled on this.

Reply:
It's also unexpected that the default is to stop/prevent. Vue has it's own convention, which is to add `.stop` and `.prevent` to the event (not the key), as 'v-on:click.stop.prevent`. No less arcane I suppose.

Reply:
But that's unhelpful here, because often you want most keys stopped, and some keys to bubble. The example is with nested menus: up/down arrows should only be handled by the active menu (not its parent), but Enter is handled by the selected item and also by the top-most menu (to close the whole stack). So the decision isn't per event type, but per key.